### PR TITLE
Add option for custom warnings guards.

### PR DESCRIPTION
**Pull request by mihaip**
Similar to custom-passes and soy-function-plugins, allows a list of class names
that will be looked up, loaded and registered.

### CODE REVIEW

**Comment by mihaip** [comment]
(I need for the demo part of https://github.com/mihaip/react-closure-compiler)

File: src/org/plovr/WarningsGuardFactory.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+   * </ul>
+   */
+  public WarningsGuard createWarningsGuard() {
+    for (Constructor<?> ctor : clazz.getConstructors()) {
+      Class<?>[] params = ctor.getParameterTypes();
+      try {
+        if (params.length == 0) {
+          return (WarningsGuard) ctor.newInstance();
+        }
+      } catch (InstantiationException e) {
```
Comment (nicks) [inline src/org/plovr/WarningsGuardFactory.java]: Throwables.propagate is usually a better choice here
https://code.google.com/p/guava-libraries/wiki/ThrowablesExplained

**Comment by nicks** [comment]
neat idea, lgtm

File: src/org/plovr/WarningsGuardFactory.java (hunk 1 above)
Comment (mihaip) [inline src/org/plovr/WarningsGuardFactory.java]: OK, see #49 (though the wiki page says that `Throwables.propagate` is not really necessary if you have access to the Java 7 multicatch statement).

**Pull request merged**